### ISSUE TITLE
How to fix warning: Metric once published using sync API should always use sync API to publish following metrics

### ISSUE THREAD

**Issue by dunalduck0** [Azure/azure-sdk-for-python#23563]
- **Package Name**: azureml-defaults
- **Package Version**: 1.36.0.post2
- **Operating System**: linux
- **Python Version**: 3.8.10

**Describe the bug**
I am running a HyperDrive job for hyperparameter search with Bayesian sampling. The metric is logged by azureml as [here](https://docs.microsoft.com/en-us/azure/machine-learning/how-to-tune-hyperparameters#specify-primary-metric-to-optimize). I am getting this warning whenever a loss is created:

03/17/2022 09:20:09 - WARNING - azureml._SubmittedRun#HD_a21b5969-dbe9-4144-b390-f3be29dcacd3_0.RunHistoryFacade.MetricsClient - Metrics Client: _execute_with_workspace_run_arguments failed ,                                retrying it after 3 seconds. Reason for failure ServiceException:
	Code: 400
	Message: (UserError) Metric once published using sync API should always use sync API to publish following metrics
	Details:

	Headers: {
	    "Date": "Thu, 17 Mar 2022 16:20:09 GMT",
	    "Content-Type": "application/json; charset=utf-8",
	    "Content-Length": "640",
	    "Connection": "keep-alive",
	    "Request-Context": "appId=cid-v1:2d2e8e63-272e-4b3c-8598-4ee570a0e70d",
	    "x-ms-response-type": "error",
	    "Strict-Transport-Security": "max-age=15724800; includeSubDomains; preload",
	    "X-Content-Type-Options": "nosniff",
	    "x-request-time": "0.042"
	}
	InnerException: {
    "additional_properties": {},
    "error": {
        "additional_properties": {
            "debugInfo": null
        },
        "code": "UserError",
        "severity": null,
        "message": "Metric once published using sync API should always use sync API to publish following metrics",
        "message_format": null,
        "message_parameters": null,
        "reference_code": null,
        "details_uri": null,
        "target": null,
        "details": [],
        "inner_error": null,
        "additional_info": null
    },
    "correlation": {
        "operation": "bf589467dcd2e9fa7a75cf5a0f224558",
        "request": "4b0b59f9f455af89"
    },
    "environment": "westus2",
    "location": "westus2",
    "time": {},
    "component_name": "metric"
}

**To Reproduce**
The job is a large GPT model fine-tuning and it's difficult to paste the entire code here. I would be much appreciated if anyone could give a hint where to look for this error. At this point, I didn't know which piece of code is causing this warning message.

**Comment by SaurabhSharma-MSFT** [comment]
@dunalduck0 Thanks for reporting this issue. We are redirecting it to services team to look into it.

**Comment by gjwoods** [comment]
@dunalduck0 - are you using MlFlow for any of your Metric logging?

MlFlow push's logs via a synchronous process, which the normal Metric logging flow does not. This is why you may be seeing the error: "message": "Metric once published using sync API should always use sync API to publish following metrics". More details incase this is the case, once a Metric has been pushed synchronously for a run, that metric cannot be pushed asynchronously for that run - aka, please make a new run to verify once MlFlow has been removed.

Please reach out to me internally for more details.

**Comment by gjwoods** [comment]
Hi @vsablok123 - this isn't an issue we can fix.
For each Metric on a Run, once the first value is sent sync or async - additional values must be sent the same way. This is a system limitation. MlFlow sends synchronously.

Generally, I recommend using MlFlow or the SDK, but not both. If you share details about your run, I can look deeper into your specifics.

**Comment by vsablok123** [comment]
Hi gjwoods, Thank you for your reply.
Your advice worked for me. I will tell in brief how I was able to resolve the above-mentioned issue?

I have removed MLFlow from the logging flow, and rely only on Native Metric tracking. 

Following advice only applies if you are working with transformers models(Huggingface).
I am using transformers model for training. The hugging-face transformers will use the MLFlow package for logging internally, so I manually disabled that by setting the env variable DISABLE_MLFLOW_INTEGRATION = TRUE

We can still log the metrics from huggingface models in AzureML Metrics by changing the Callback to **AzureMLCallback** as a parameter in Trainer Object. 
 See https://github.com/huggingface/transformers/blob/main/src/transformers/integrations.py for more details on this.
